### PR TITLE
add the Datree plugin to the repository

### PR DESCRIPTION
**Pull request by hadar-co**
Tested locally: :)
```
Installing plugin: datree
Installed plugin: datree
\
 | Use this plugin:
 | 	kubectl datree
 | Documentation:
 | 	https://github.com/datreeio/kubectl-datree
/
```

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Comment by hadar-co** [comment]
CLA has been signed, please review again:)
@k8s-ci-robot

**Comment by hadar-co** [comment]
@ahmetb @chriskim06 
Hi :)
Please assist in advancing this PR to completion. I have signed the CLA, how do I prompt a retest?
Thanks!

**Comment by ahmetb** [comment]
I've left a comment, is it not visible to you?

**Comment by hadar-co** [comment]
@ahmetb
My bad, I missed it.. thanks :)

**Comment by hadar-co** [comment]
@ahmetb 
I realized that my user was not linked to my commit, I fixed that.
Now the manifest validation produces this error:

I0202 14:46:21.358190    1983 downloader.go:54] Extracting zip archive to "/tmp/krew-downloads934326152"
I0202 14:46:21.358522    1983 install.go:101] Deleting the download staging directory /tmp/krew-downloads934326152
	W0202 14:46:21.358700    1983 install.go:164] failed to install plugin "datree": install failed: failed to unpack into staging dir: failed to unpack the plugin archive: failed to extract file: can't create file in zip destination dir: open /tmp/krew-downloads934326152/__MACOSX/._LICENSE: no such file or directory

Any idea why this happens? Locally the installation works.
Thanks for the help :)

**Comment by ahmetb** [comment]
> Scan your k8s files in cluster

But there are no files within the cluster?

Does this call the Kubernetes apiserver at all or is it a static file validator? If so it shouldn't be a kubectl plugin probably.

**Review by ahmetb — CHANGES_REQUESTED** [review]
sorry I realized github mobile app didn't publish this

**Comment by ahmetb** [comment]
I think both your website and your plugin description is doing a suboptimal job explaining what the plugin does.

There are no "files" or "manifests" in a cluster. They live on your filesystem. Once it reaches the cluster, it's a resource/API object. That's why right now your documentation and plugin description sound quite odd to me. I thought it scans my local files and matches them with the corresponding live resources on the cluster, but apparently that's not it.

I'll let you iterate on those, otherwise we'd be doing disservice to users. Also words like k8s are redundant in the short description.

**Comment by hadar-co** [comment]
@ahmetb 
Thanks for the comments, I definitely agree that the wording was unclear.
I fixed it and recommitted.

The plugin validation is still failing with the error I pasted in my last comment, can you please assist?
Thank you

File: plugins/datree.yaml
Comment (ahmetb) [inline plugins/datree.yaml:12]: If empty, omit.

File: plugins/datree.yaml
Comment (ahmetb) [inline plugins/datree.yaml:8]: This is too long. Should be 50 max.

**Review by ahmetb — CHANGES_REQUESTED** [review]

File: plugins/datree.yaml
Comment (ahmetb) [inline plugins/datree.yaml:10]: k8s → Kubernetes

It's unclear what "the tool" here is.

**Comment by hadar-co** [comment]
@ahmetb 
Fixed :)

**Comment by ahmetb** [comment]
Two points

1. right now the tests are failing because of a bug we fixed, try this:

       git fetch origin
       git rebase origin/master
       git push -f

2. your plugin installs random stuff to users' computer, which is not what any user wants, or any program you download from a package manager would do, and we can't accept the plugin like that:

    ```sh
      # Install datree if it isn't installed
      if ! command -v datree &> /dev/null; then
          echo "Datree is not installed on your machine, proceeding to install..."
          curl https://get.datree.io | /bin/bash
      fi
    ```

    I can recommend (1) not distributing the external tool, many other plugins do this and asks users to install additional programs in caveats section (2) bundle the external tool in your plugin archive and extract it and call to it. it doesn't have to be in $PATH to be called by your plugin, you can find its path by looking at the currently executing executable's path and resolving its symlink (3) make your main executable the plugin binary, and make it run as a kubectl plugin by looking at os.Argv[0] and observing the kubectl- prefix.

As it stands today, plugins installing other programs to the system are not acceptable.
/hold

**Comment by hadar-co** [comment]
@ahmetb 
Got it, I went with suggestion (1) :)

**Comment by hadar-co** [comment]
@ahmetb 
Hi, is something still blocking the approval?
Thanks

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**
